src/modules/endpoint/models/CategoriesContainer.js

```diff
@@ -0,0 +1,72 @@
+import normalizeUrl from 'normalize-url'
+
+/**
+ * Contains a Map [string -> CategoryModel] and some helper functions
+ */
+class CategoriesContainer {
+  /**
+   * Creates a Map [url -> category] from the categories provided,
+   * whose parent attributes are first changed from id to url
+   * @param categories CategoryModels as array
+   */
+  constructor (categories = []) {
+    this._categories = new Map(categories.map(category => ([category.url, category])))
+  }
+
+  /**
+   * @return {CategoryModel[]} categories The categories as array
+   */
+  toArray () {
+    return Array.from(this._categories.values())
+  }
+
+  /**
+   * Returns the category with the given url
+   * @param url The url
+   * @return {CategoryModel | undefined} The category
+   */
+  getCategoryByUrl (url) {
+    return this._categories.get(normalizeUrl(url))
+  }
+
+  /**
+   * Returns the category with the given id
+   * @param id The url
+   * @return {CategoryModel | undefined} The category
+   */
+  getCategoryById (id) {
+    return this.toArray().find(category => category.id === Number(id))
+  }
+
+  /**
+   * Returns all children of the given category
+   * @param category The category
+   * @return {CategoryModel[] | undefined} The children
+   */
+  getChildren (category) {
+    if (category) {
+      return this.toArray()
+        .filter(_category => _category.parentUrl === category.url)
+        .sortBy(_category => _category.order)
```

Comment:
Sorry, my bad :/ .sortBy is not part of ES6, you need to use
`.sort((category1, category2) => category1.order - category2.order)`

Reply:
Maybe you can include the sort functionality in a test of CategoriesContainer? 🤔

Reply:
no, I should be more carefully and testing the stuff I change :/
yeah sure, I'll add a test for this